### ISSUE TITLE
Organizzazione del lavoro e suddivisione dei compiti

### ISSUE THREAD

**Issue by Kavuti** [sorcio/python-italia-telegram-bot#5]
Ciao ragazzi!
Sento il bisogno di aprire la Issue per organizzarci con il lavoro. Prima di tutto credo che sia utile che ognuno si occupi solo di alcune cose, senza il rischio di accavallarsi a modificare le stesse funzionalità in più persone se non strettamente necessario.

Pensavo che potremmo dividerci il lavoro per task o per moduli. Una suddivisione di esempio (non prendetela sul serio) potrebbe essere:
- Christian: mangiare mele
- Davide: predisposizione canale e implementazione invio messaggi sul canale
- Mattia: warn, kick, ban
- basic-ph (aspettiamo un nome): tagliare prato
ecc... (Non so quanti altri sono coinvolti)

Una suddivisione per modulo invece potrebbe essere (sempre di esempio):
- Christian: modulo admin
- Davide: modulo afk
- Mett: modulo ban
- basic: modulo vattelapesca

Che ne pensate? Condividete il fatto che potrebbe generarsi confusione se non ci organizziamo quanto prima?

**Comment by sorcio** [comment]
Christian, ti va di delineare quali siano i compiti, prima di assegnarli a persone specifiche? Se li formuliamo bene in forma di issue su GH allora il problema della suddivisione diventa secondario, e possiamo accogliere contributi anche di altre persone oltre alle quattro elencate nei tuoi esempi :)

Inoltre io a tempo perso continuerò a lavorare sulla mia implementazione del bot (quella nel branch `tgtrio` di questo repo) :)

**Comment by Kavuti** [comment]
Vista l'organizzazione naturale che stanno acquisendo le Issues da te aperte, proporrei di dividere tra i membri le risoluzioni delle singole Issue (una volta che saranno chiarite ovviamente). Attualmente sono presenti diverse Issue:
- Anti spam
- Protezione da flood
- Onboarding di nuovi utenti
- Errore nel modulo **markdownhelp**
- Gestione provvedimenti quali warn, kick e ban.

Per il momento eviterei di mettere altra carne al fuoco, visto che il lavoro, sia intellettuale che pratico, è già impegnativo vista la precisione e la complessità di implementazione di cui abbiamo bisogno. Attendo un vostro parere. ✌

**Comment by Kavuti** [comment]
Ciao a tutti
@sorcio @iflare3g @MattiaFailla @basic-ph
Prima di dividerci i compiti, sto notando che c'è confusione con la gestione del codice del bot. Mi pare che attualmente ci siano almeno 3 repository diverse contenenti codici probabilmente disallineati. Anche le Issue, che si trovano attualmente tutte su questa repository, probabilmente sono nel posto sbagliato. Potremmo riunire tutto il lavoro in un'unica repository (qualunque sia, anche una nuova)? Potrebbe anche essere privata se necessario, però in questo momento l'impressione che ho è che non si sa da quale repository si andrà a prendere il codice finale del bot. 
Fatemi sapere che ne pensate, se avete proposte, ecc...

**Comment by iflare3g** [comment]
Concordo assolutamente e , come anticipato su TG, sarebbe utile utilizzare il project di GH in modo da avere una lista di cose da fare in modo che tutti possano sempre essere informati sullo stato delle cose e con le label dovremmo anche riuscire a distinguere i vari topic ( anti spam, ecc ecc )

**Comment by basic-ph** [comment]
Mi sembra un'ottima idea per ottimizzare il lavoro, attendo indicazioni

**Comment by MattiaFailla** [comment]
Il problema è che, appunto, ci sono tre versioni del progetto con sorgenti differenti. Se pensate che tgbot faccia al caso nostro sarebbe meglio se spostassimo le risorse sulla repo di @Kavuti così da andare avanti su una singola linea.

Potremmo usare questa repo per discutere su cose più generali e concentrare gli issue veri e propri su quella di tgbot.

Fatemi sapere cosa ne pensate.

**Comment by sorcio** [comment]
L'idea di questo repo è, appunto, parlare del "progetto bot per Python Italia" in generale, a prescindere da ogni singola implementazione. Poi se le discussioni sulla soluzioni tgbot si svolgono qui o altrove per me cambia poco. Io sono +1 a concentrare tutto qui, per quello che conta, ma non mi oppongo alla proposta di @MattiaFailla 

Idem per il project, se serve definire pianificazione e milestone si può fare qui o altrove.

**Comment by iflare3g** [comment]
> Il problema è che, appunto, ci sono tre versioni del progetto con sorgenti differenti. Se pensate che tgbot faccia al caso nostro sarebbe meglio se spostassimo le risorse sulla repo di @Kavuti così da andare avanti su una singola linea.
> 
> Potremmo usare questa repo per discutere su cose più generali e concentrare gli issue veri e propri su quella di tgbot.
> 
> Fatemi sapere cosa ne pensate.

Assolutamente d'accordo, infatti la PR comunque l'avrei fatta sul fork di @Kavuti
